### PR TITLE
Fix order of registered frees in loop fuser to prevent frees occurrin…

### PR DESCRIPTION
**Pull request by liu15**
…g before all actions are flushed

### CODE REVIEW

**Review by robinson96 — APPROVED** [review]
LGTM

File: src/care/LoopFuser.h [hunk 1]
```diff
@@ -374,9 +374,15 @@ class FusedActionsObserver : public FusedActions {
       for (auto & priority_action : m_fused_action_order) {
          FusedActions * const & actions = priority_action.second;
          if (actions -> actionCount() > 0) {
-            actions->flushActions(async, fileName, lineNumber);
+            if (verbose) {
+               printf("flushing actions at priority %g\n", priority_action.first);
+            }
+            actions->flushActions(true, fileName, lineNumber);
```
Comment (adayton1) [inline src/care/LoopFuser.h:380]: Shouldn't you be using `async` instead of `true`?

Comment (liu15) [reply src/care/LoopFuser.h:380]: @robinson96 was this change intentional or leftover from debugging?

Comment (adayton1) [reply src/care/LoopFuser.h:380]: OK, that makes more sense.

**Review by adayton1 — APPROVED** [review]
LGTM

File: src/care/LoopFuser.h (hunk 1 above)
Comment (adayton1) [reply src/care/LoopFuser.h:380]: I would just add a comment to the code explaining this.

**Review by adayton1 — APPROVED** [review]
LGTM

**Pull request merged**